### PR TITLE
WIP: Added starting point for Jacoco coverage parser

### LINKED ISSUE #128: Support for jacoco code coverage
One of the steps in our pipeline is enforcing code coverage metrics on the changed code. The way jacoco enforces those metrics is by having a task that passes/fails across the whole code. We were able to implement what we wanted with the help of the violations lib, with a twist:

1. violation lib does not understand jacoco reports, so we wrote an extra step that converts the coverage xml to a code climate json, raising an issue for every method that is more than x lines long and has a coverage of less than y%
2. we used violation lib plugins to read that report and post comments on the MR about methods with insufficient coverage (it is a bit hard to post at the individual line level, not from a technical perspective, but from a usability one)

I was thinking of contributing the jacoco report but I do not see a way of configuring parsers. I could contribute with my settings, but probably others would have different uses. What's your take on situations like this?

### PR DESCRIPTION
**Pull request by ateglas**
Closes #128 

Starting point for the coverage parser. It basically takes the jacocoTestReport.xml file and produces a set of violations that can be cross checked afterwards against the VCS code changes. 

Definitely not ready to release, just posting this for early feedback. It still needs:
- unit tests (did not had time to port those as well)
- finding the actual source file (we only have package and file name, these can be hidden inside various folder structures like */src/main/java/...)
- sample usage in the supporting pages

### CODE REVIEW

**Comment by tomasbjerre** [comment]
I think it looks good.

I think you should get the `filename` when from replacing dots with slashes. Similar to what is done here:
https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/parsers/JUnitParser.java#L122
The important thing is that it needs to be that last part of the path + filename. So that this code matches:
https://github.com/tomasbjerre/violation-comments-lib/blob/master/src/main/java/se/bjurr/violations/comments/lib/ChangedFileUtils.java#L25
There are currently no file operations in any parser, and I'd like to keep it that way.

You don't need to bother with documentation. It is OK to have the configuration hard coded in this PR. If you add unit test and implement the filename, I can take it from there.

File: src/main/java/se/bjurr/violations/lib/parsers/JacocoXmlParser.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+public class JacocoXmlParser implements ViolationsParser {
+
+    private static final int MIN_LINE_COUNT = 4;
+
+    private static final double MIN_COVERAGE = 0.7;
+
+    @Override
+    public Set<Violation> parseReportOutput(String reportContent, ViolationsLogger violationsLogger) throws Exception {
+        Set<Violation> violations = new HashSet<>();
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
```
Comment (tomasbjerre) [inline src/main/java/se/bjurr/violations/lib/parsers/JacocoXmlParser.java:34]: The current xml-parsers use `XmlStreamReader`, I think I would prefer using that one everywhere.
https://github.com/tomasbjerre/violations-lib/blob/5fa2ec7c1852cd9749309b2ae5309ef9a7c30b9f/src/main/java/se/bjurr/violations/lib/parsers/FindbugsParser.java#L54

Comment (ateglas) [reply src/main/java/se/bjurr/violations/lib/parsers/JacocoXmlParser.java:34]: I saw that XMLStreamReader was being used, for me the DOMParser was the easiest to publish something for feedback. Will change to the reader together with the tests.

**Comment by ateglas** [comment]
There's no need to replace dots, it already contains slashes in the jacoco report for the package name. What I did not knew was the ChangedFileUtils part. I always thought that it needs to be a full match, my conversion from Jacoco to Codeclimate format was always going to all source directories to locate the actual file. Now I see that it just needs to match the end part, that simplifies things a lot.

**Comment by ateglas** [comment]
It is a bit late so I might have messed it up, but should be closer to what we discussed now.

**Comment by ateglas** [comment]
Is there something I can do about the openjdk8 failure? Seem like it is related to googleJavaFormat and java11, not something from my PR.

**Pull request merged**